contracts/0.4.24/StETH.sol

```diff
@@ -403,6 +458,9 @@ contract StETH is IERC20, Pausable {
         uint256 accountShares = shares[_account];
         require(_sharesAmount <= accountShares, "BURN_AMOUNT_EXCEEDS_BALANCE");
 
+        uint256 amount = getPooledEthByShares(_sharesAmount);
```

Comment:
The calculation of the amount of burned stETH isn't correct in this line. Cause the burning also decreases the total amount of minted shares, the balance of the account will be decreased by a different amount, so we can't just calculate it as `sharesAmount * pooledEth / totalAmountOfShares`. The simplest correct way to count the burned amount of stETH is `balanceBefore - balanceAfter`.

Reply:
I like your idea :+1: . Though, the `SharesBurnt` event has the following definition
```
     * @param amount amount of burnt shares (expressed in stETH just before burning invocation)
```
So it doesn't violate the definition, but your approach sounds more practical.

Reply:
Maybe rename `amount` to `tokenAmountBeforeBurn`? To make it more semantically clear?

Reply:
I'd go with what Eugene suggested

the amount after the burn is easily recoverable via `getPooledEthByShares(event.sharesAmount)`, but the amount before burn cannot be easily obtained

Reply:
we can add `tokenAmountAfterBurn`, though

I'd even call these `preRebaseTokenAmount` and `postRebaseTokenAmount`; this will also make it more clear that rebase is happening here

Reply:
+1 on `preRebaseTokenAmount` and `postRebaseTokenAmount`! I like it!

Reply:
The main advantage here is that is more usable from subgraph perspective